metastore/src/java/org/apache/hadoop/hive/metastore/SerDeStorageSchemaReader.java

```diff
@@ -47,10 +48,10 @@
       Deserializer s = HiveMetaStoreUtils.getDeserializer(conf, tbl, false);
       return HiveMetaStoreUtils.getFieldsFromDeserializer(tbl.getTableName(), s);
     } catch (Exception e) {
-      StringUtils.stringifyException(e);
-      throw new MetaException(e.getMessage());
+      throw new MetaException(StringUtils.stringifyException(e));
```

Comment:
Why does this throw? MetaException?  Try something else?

Reply:
If the serde of the table is not a part of `metastore.serdes.using.metastore.for.schema`,  like JsonSerDe(openx).  MetaStore will use the defined serde to get the columns of the table. It's may be better to tell the caller that the exception happens when retrieving the columns from serde other than return null or an empty list, so the GetColumnsOperation can bring the exception to the front.

Reply:
Hi, @belugabehr could you please take another look at the changes? thank you

Reply:
Hey @belugabehr, could you please take another look? thanks a lot.